fdb-relational-grpc/src/main/java/com/apple/foundationdb/relational/jdbc/TypeConversion.java

```diff
@@ -251,23 +252,36 @@ private static Struct toStruct(RelationalStruct relationalStruct) throws SQLExce
     public static Object fromColumn(int columnType, Column column) throws SQLException {
         switch (columnType) {
             case Types.ARRAY:
+                checkColumnType(columnType, column.hasArray());
                 return fromArray(column.getArray());
             case Types.BIGINT:
+                checkColumnType(columnType, column.hasLong());
```

Comment:
Does this work correctly if the column has the default value for the desired item, i.e. `0` for a long, or an int, or an empty string?

Reply:
I believe so. The value of the column in case you try to `get` an unpopulated field is the default value (no error) hence we need to test that the value is populated. If there was no value populated then this should fail (we don't want to assume a default value here, there should be an explicit value). Was that what you asked?
Testing with numerical default values (0) also succeeds, the check passes when setting a default value.